src/content-helper/editor-sidebar/tabs/sidebar-tools-tab.tsx

```diff
@@ -31,49 +33,54 @@ type SidebarToolsTabProps = {
  * @param {SidebarToolsTabProps} props The component's props.
  */
 export const SidebarToolsTab = (
-	{ trackToggle }: Readonly<SidebarToolsTabProps>
+	{ trackToggle, permissions }: Readonly<SidebarToolsTabProps>
 ): React.JSX.Element => {
 	const { settings, setSettings } = useSettings<SidebarSettings>();
 
 	return (
 		<Panel>
-			<PanelBody
-				title={ __( 'Title Suggestions (Beta)', 'wp-parsely' ) }
-				initialOpen={ settings.TitleSuggestions.Open }
-				onToggle={ ( next ) => {
-					setSettings( {
-						TitleSuggestions: {
-							...settings.TitleSuggestions,
-							Open: next,
-						},
-					} );
-					trackToggle( 'title_suggestions', next );
-				} }
-			>
-				<VerifyCredentials>
-					<TitleSuggestionsPanel />
-				</VerifyCredentials>
-			</PanelBody>
+			{ permissions.TitleSuggestions &&
+				<PanelBody
+					title={ __( 'Title Suggestions (Beta)', 'wp-parsely' ) }
+					initialOpen={ settings.TitleSuggestions.Open }
+					onToggle={ ( next ) => {
+						setSettings( {
+							TitleSuggestions: {
+								...settings.TitleSuggestions,
+								Open: next,
+							},
+						} );
+						trackToggle( 'title_suggestions', next );
+					} }
+				>
+					<VerifyCredentials>
+						<TitleSuggestionsPanel />
+					</VerifyCredentials>
+				</PanelBody>
+			}
 
-			<PanelBody
-				title={ __( 'Smart Linking (Beta)', 'wp-parsely' ) }
-				initialOpen={ settings.SmartLinking.Open }
-				onToggle={ ( next ) => {
-					setSettings( {
-						SmartLinking: {
-							...settings.SmartLinking,
-							Open: next,
-						},
-					} );
-					trackToggle( 'smart_linking', next );
-				} }
-			>
-				<VerifyCredentials>
-					<SmartLinkingPanel
-						context={ SmartLinkingPanelContext.ContentHelperSidebar }
-					/>
-				</VerifyCredentials>
-			</PanelBody>
+			{ permissions.SmartLinking &&
```

Comment:
This is a suggestion for the future, not really necessary for now, but instead of doing these conditional checks, we can likely create a `<ValidatePermissions>` component, and have it validate the correct permissions for each panel.

Reply:
Yes, I think this is a good idea, and in line with what we're doing for credential validation. 🙂